compiler/src/iree/compiler/Dialect/Stream/Transforms/FoldUniformOperands.cpp

```diff
@@ -35,6 +36,54 @@ namespace {
 // Per-dispatchable export optimization
 //===----------------------------------------------------------------------===//
 
+// Returns the uniformly duplicated values across all indexing map entries of
+// duplicated values and a bit vector indicating the 'dead' values.
+//
+// Example:
+// Map of duplicated values per entry: [[0, 1, 0, 1], [0, 1, 0, 2]]
+// Returns:
+// - A map of uniformly duplicated values: [0, 1, 0, 2]
+// - A bit vector with dead values: b0010
+static std::pair<SmallVector<unsigned>, llvm::BitVector>
+getUniformDupeIndexingAndDeadValues(
+    const SmallVector<SmallVector<unsigned>> &dupeIndexMaps) {
+  if (dupeIndexMaps.size() == 0) {
+    return {};
+  }
+  unsigned numValues = dupeIndexMaps[0].size();
+  llvm::BitVector sameValues(numValues);
+  llvm::BitVector deadOperandsMap(numValues);
+  SmallVector<unsigned> uniformDupeIndexMap =
+      llvm::to_vector(llvm::seq(0u, numValues)); // old -> new
+  for (unsigned idx = 0; idx < numValues; ++idx) {
+    if (deadOperandsMap.test(idx))
+      continue;
+    // Each bit represents a value that duplicates the value at idx.
+    // We walk all the sites and AND their masks together to get the safe
+    // set of duplicate operands.
+    // Example for %0: (%a, %b, %a) -> b001
+    // Example for %1: (%a, %b, %a) -> b000
```

Comment:
I find it easier to understand if we replace it with below. `%0` and `%1` are quite ambiguous to me, because I don't find any `%0` and `%1` in other places.


```suggestion
    // Example for idx=0 and list=[(%a, %b, %a)] -> b001
    // Example for idx=1 and list=[(%a, %b, %a)] -> b000
```

Reply:
Yeah, that's better.